### ISSUE TITLE
backend offline_simulator_no_noise/noise does not keep the measurement bit order

### ISSUE THREAD

**Issue by naitsirhcremmos** [qiskit-community/qiskit-aqt-provider#85]
### Informations

- **Qiskit AQT Provider version**: v0.16.0
- **Python version**: 3.9.5
- **Operating system**: Windows

### What is the current behavior?
The result after a measurement comes out in the wrong bit order when the qubit order is permuted (classical bits for readout keep the former order).
Permuted order, wrong result (```1001, 0000``` expected)
![Offline_sim_issue_2](https://github.com/qiskit-community/qiskit-aqt-provider/assets/126893545/b1ce3a16-1aae-4259-89dd-2eed187839c9)

### Steps to reproduce the problem
Run a circuit like 
```
nQ = 4
qcirc = QuantumCircuit(nQ)
qcirc.h(0)
qcirc.cx(0, 3)
qcirc.measure_all()
```
and change the arrangement order either with ```initial_layout``` in ```transpile``` or by 
``` 
permute = sample(range(0, num_qubits), num_qubits)
qc = QuantumCircuit(num_qubits, num_qubits)
perm_qc = qc.compose(circ, qubits=permute)
result = execute(new_circ, backend, shots=num_shots).result()
```
### What is the expected behavior?

If the order is the initial one, correct result

![Offline_sim_issue_1](https://github.com/qiskit-community/qiskit-aqt-provider/assets/126893545/d661359d-5373-4036-a80d-8a50e90b70e8)

else the result will follow the previously stated behavior

**Comment by ballodominikistvan** [comment]
Maybe something is wrong in the CX function about indexing the qubits in the circuit, you should check it.

**Comment by airwoodix** [comment]
Thanks for the bug report.

I can reproduce the mismatch between the Aer simulator and the offline simulators from `qiskit_aqt_provider`. This is due too naive memory model used in this package and will be fixed together with #80.

Reproduction code:

```python
import qiskit
from qiskit import QuantumCircuit
from qiskit_aer import AerProvider

from qiskit_aqt_provider import AQTProvider

aer = AerProvider().get_backend("aer_simulator")
sim = AQTProvider("token").get_backend("offline_simulator_no_noise")
sim.options.with_progress_bar = False

base = QuantumCircuit(3)
qc = QuantumCircuit(3)
qc.h(0)
qc.cx(0, 2)
qc.measure_all()

perm_qubits = base.compose(qc, qubits=[0, 2, 1])
perm_all = base.compose(qc, qubits=[0, 2, 1], clbits=[0, 2, 1])

print(set(qiskit.execute(perm_qubits, sim).result().get_counts()))
print(set(qiskit.execute(perm_all, sim).result().get_counts().keys()))

print(set(qiskit.execute(perm_qubits, aer).result().get_counts()))
print(set(qiskit.execute(perm_all, aer).result().get_counts()))
```

```
{'000', '011'}
{'000', '011'}
{'000', '101'}
{'000', '011'}
```
